src/purchases.ts

```diff
@@ -423,41 +437,79 @@ export default class Purchases {
    * @param {UpgradeInfo} upgradeInfo Android only. Optional UpgradeInfo you wish to upgrade from containing the oldSKU
    * and the optional prorationMode.
    * @param {String} type Optional type of product, can be inapp or subs. Subs by default
-   * @returns {Promise<{ productIdentifier: string, customerInfo:CustomerInfo }>} A promise of an object containing
-   * a customer info object and a product identifier. Rejections return an error code,
-   * a boolean indicating if the user cancelled the purchase, and an object with more information. The promise will
-   * also be rejected if configure has not been called yet.
+   * @param {boolean} googleIsPersonalizedPrice Android and Google only. Optional boolean indicates personalized pricing on products available for purchase in the EU.
+   * For compliance with EU regulations. User will see "This price has been customize for you" in the purchase dialog when true.
+   * See https://developer.android.com/google/play/billing/integrate#personalized-price for more info.
+   * @deprecated, use purchaseStoreProduct instead
    */
   public static async purchaseProduct(
     productIdentifier: string,
     upgradeInfo?: UpgradeInfo | null,
-    type: PURCHASE_TYPE = PURCHASE_TYPE.SUBS
+    type: PURCHASE_TYPE = PURCHASE_TYPE.SUBS,
   ): Promise<MakePurchaseResult> {
     await Purchases.throwIfNotConfigured();
     return RNPurchases.purchaseProduct(
       productIdentifier,
       upgradeInfo,
       type,
+      null,
+      null,
       null
     ).catch((error: PurchasesError) => {
       error.userCancelled = error.code === PURCHASES_ERROR_CODE.PURCHASE_CANCELLED_ERROR;
       throw error;
     });
   }
 
+  /**
+   * Make a purchase
+   *
+   * @param {PurchasesStoreProduct} product The product you want to purchase
+   * @param {GoogleProductChangeInfo} googleProductChangeInfo Android only. Optional GoogleProductChangeInfo you 
+   * wish to upgrade from containing the oldProductIdentifier and the optional prorationMode.
+   * @param {boolean} googleIsPersonalizedPrice Android and Google only. Optional boolean indicates personalized pricing on products available for purchase in the EU.
+   * For compliance with EU regulations. User will see "This price has been customize for you" in the purchase dialog when true.
+   * See https://developer.android.com/google/play/billing/integrate#personalized-price for more info.
+   * @returns {Promise<{ productIdentifier: string, customerInfo:CustomerInfo }>} A promise of an object containing
+   * a customer info object and a product identifier. Rejections return an error code,
+   * a boolean indicating if the user cancelled the purchase, and an object with more information. The promise will
+   * also be rejected if configure has not been called yet.
+   */
+  public static async purchaseStoreProduct(
+    product: PurchasesStoreProduct,
+    googleProductChangeInfo?: GoogleProductChangeInfo | null,
+    googleIsPersonalizedPrice?: boolean | null,
```

Comment:
Should we make `googleProductChangeInfo` and `googleIsPersonalizedPrice` just `?` instead of also `| null`? Check this out https://basarat.gitbook.io/typescript/recap/null-undefined#final-thoughts

I know we are a bit all over the place in the code, but since this is a new function, I think it's a good opportunity to follow best practices

Same comment would apply to other new parameters and functions in this file

Reply:
Oh! Either those `| null| or `?` might have been left in there by accident 😬 

I don't think they either of them affect the call signature since order does matter but I could probably just have it be the `?` 😇

Reply:
but only `?` won't allow for passing `googleIsPersonalizedPrice` without passing  `googleProductChangeInfo` so I think what it is right now `null | or ?` makes more sense, so it can be called like

```
const subscriptionOptionResult2: MakePurchaseResult = await Purchases.purchaseSubscriptionOption(
    subscriptionOption,
    null,
    googleIsPersonalizedPrice
  );
```